frontends/infinite-corridor/src/components/forms/EditFieldAppearanceForm.test.tsx

```diff
@@ -0,0 +1,68 @@
+import {
+  renderTestApp,
+  screen,
+  setMockResponse,
+  fireEvent
+} from "../../test-utils"
+import * as factory from "../../api/fields/factories"
+import { urls } from "../../api/fields"
+
+describe("EditFieldAppearanceForm", () => {
+  it("Displays the field title, appearance inputs", async () => {
+    const field = factory.makeField({ is_moderator: true })
+    setMockResponse.get(urls.fieldDetails(field.name), field)
+    expect(field.is_moderator).toBeTruthy()
+    renderTestApp({ url: `${urls.fieldDetails(field.name)}manage/#appearance` })
+    const descInput = (await screen.findByLabelText(
+      "Description"
+    )) as HTMLInputElement
+    const titleInput = (await screen.findByLabelText(
+      "Title"
+    )) as HTMLInputElement
+    expect(titleInput.value).toEqual(field.title)
+    expect(descInput.value).toEqual(field.public_description)
+  })
+
+  it("Shows an error if a required field is blank", async () => {
+    const field = factory.makeField({ is_moderator: true })
+    setMockResponse.get(urls.fieldDetails(field.name), field)
+    renderTestApp({ url: `${urls.fieldDetails(field.name)}manage/#appearance` })
+    const titleInput = await screen.findByLabelText("Title")
+    const titleError = screen.queryByText("Title is required")
+    expect(titleError).toBeNull()
+    fireEvent.change(titleInput, {
+      target: { value: "" }
+    })
+    fireEvent.blur(titleInput)
```

Comment:
@mbertrand One more "by the way" comment: I usually trigger events via `user`:

```ts
import user from "@testing-library/userEvent" // test-utils also exports this as `{ user }` for convenience.

user.click(htmlElement)
user.type("blah blah blah{Enter}")
```
 
The `user` interface is a higher-level API that more closely resembles real user interaction. (`@testing-library`'s general philosophy is "write tests that resemble user interaction".)

For example, `user.type` types the text (character by character, I think... which is still plenty fast). And `user.click` fires most (all?) the events that would be fired by a click: pointerdown, pointermove, pointerup, click.

Anyway, that's why I generally use those methods though I suspect it doesn't usually matter.

**BTW:** `user.type` can be slightly annoying because it's like typing...you have to clear the text first:

```ts
input.setSelectionRange(0, input.value.length) // select all the text
user.type(input, 'whatever new thing you want') // typing clears the selected text
```
https://github.com/testing-library/user-event/issues/232 

cc @abeglova

Reply:
Oh! I remember why `user.click` is nice for clicking... it does not trigger events if the element is disabled, whereas fireEvent just triggers the event no matter what.

Related...
- https://github.com/jsdom/jsdom/issues/2665 _Closed because apparently this follows some spec, and browsers do not follow the spec. 🤷 _
- https://github.com/testing-library/dom-testing-library/issues/92

Reply:
Thanks, I noticed that `user.type(input, text)` throws an error if `text` is blank ("") - `Expected key descriptor but found "" in ""` - so I left that test as is but changed the others.